databroker/eventsource/archiver.py

```diff
@@ -10,45 +10,113 @@
 import pytz
 import uuid
 
+import numpy as np
+
 import requests
 import pandas as pd
 
+from ..utils import ALL
 
 class ArchiverEventSource(object):
-    def __init__(self, url, timezone, pvs):
+    
+    def __init__(self, config):
         """
-        DataBroker plugin
-
-        Parameters
-        ----------
-        url : string
-            e.g., 'http://host:port/'
-        timezone : string
-            e.g., 'US/Eastern'
-        pvs : dict
-            a dict mapping human-friendly names to PVs
+        Shim class to turn the EPICS Archiver Appliance into EventSource
         """
+
+        if all (k in config for k in ['name', 'url']):
+            pass
+        else:
+            raise TypeError("config {} does not include one of required"
+                            " keys (name, url).".format(config))
+        
+        self._name = config['name']
+        
+        url =  config['url']
         if not url.endswith('/'):
             url += '/'
         self.url = url
         self.archiver_addr = self.url + "retrieval/data/getData.json"
+
+        if 'timezone' in config:
+            timezone = config['timezone']
+        else:
+            timezone = 'US/Eastern'
         self.tz = pytz.timezone(timezone)
+
+        if 'pvs' in config:
+            pvs = config['pvs']
+        else :
+            pvs = {}
         self.pvs = pvs
+        
         self._descriptors = {}
 
+
+    @property
+    def name(self):
+        return self._name
+
+    def add_pvs(self, new_pvs):
+        self.pvs.update(dict(new_pvs))
+        return dict(self.pvs)
+
+    def get_pvs(self):
+        return dict(self.pvs)
+
     def insert(self, name, doc):
-        raise NotImplementedError()
+         """
+         Not supported, data archiving is managed via the EPICS Archiver Appliance Toolkit
+         """
+         raise TypeError("Not supported, data archiving is managed via the EPICS Archiver Appliance Toolkit")
 
     def stream_names_given_header(self, header):
         # We actually don't use the header in this case.
-        return ['archiver_{}'.format(name) for name in self.pvs]
+        """
+        Return a list of user-defined PV names
+
+        Parameters
+        ----------
+        header: Header
```

Comment:
This docstring won't render properly in HTML without a space before the colon, like `header : Header`.